src/devices/Ws28xx/samples/Ws28xx.Sample.cs

```diff
@@ -13,47 +14,157 @@ internal class Program
     {
         // Configure the count of pixels
         private const int Count = 8;
-
+        private static Ws28xx s_neo;
         public static void Main()
+        {
+            Console.Clear();
+            Console.CancelKeyPress += new ConsoleCancelEventHandler(CancelHandler);
+            Run();
+        }
+
+        private static void Run()
         {
             var settings = new SpiConnectionSettings(0, 0)
             {
-                ClockFrequency = 2_400_000, Mode = SpiMode.Mode0, DataBitLength = 8
+                ClockFrequency = 2_400_000,
+                Mode = SpiMode.Mode0,
+                DataBitLength = 8
             };
-
-            // Create a Neo Pixel x8 stick on spi 0.0
             var spi = SpiDevice.Create(settings);
 
 #if WS2808
-            var neo = new Ws2808(spi, count);
+            s_neo = new Ws2808(spi, count);
 #else
-            var neo = new Ws2812b(spi, Count);
+            s_neo = new Ws2812b(spi, Count);
 #endif
+            while (true)
+            {
+                ColorWipe(s_neo, Color.White, Count);
+                ColorWipe(s_neo, Color.Red, Count);
+                ColorWipe(s_neo, Color.Green, Count);
+                ColorWipe(s_neo, Color.Blue, Count);
 
-            // Display basic colors for 5 sec
-            BitmapImage img = neo.Image;
-            img.Clear();
-            img.SetPixel(0, 0, Color.White);
-            img.SetPixel(1, 0, Color.Red);
-            img.SetPixel(2, 0, Color.Green);
-            img.SetPixel(3, 0, Color.Blue);
-            img.SetPixel(4, 0, Color.Yellow);
-            img.SetPixel(5, 0, Color.Cyan);
-            img.SetPixel(6, 0, Color.Magenta);
-            img.SetPixel(7, 0, Color.FromArgb(unchecked((int)0xffff8000)));
-            neo.Update();
-            System.Threading.Thread.Sleep(5000);
-
-            // Fade in first pixel
-            byte b = 0;
+                TheatreChase(s_neo, Color.White, Count);
+                TheatreChase(s_neo, Color.Red, Count);
+                TheatreChase(s_neo, Color.Green, Count);
+                TheatreChase(s_neo, Color.Blue, Count);
+
+                Rainbow(s_neo, Count);
+                RainbowCycle(s_neo, Count);
+                TheaterChaseRainbow(s_neo, Count);
+
+            }
+
+        }
+
+        private static void CancelHandler(object sender, ConsoleCancelEventArgs e)
+        {
+            BitmapImage img = s_neo.Image;
             img.Clear();
-            while (true)
+            s_neo.Update();
```

Comment:
NIT: should we also add Dispose code here in the Cancellation handler?

Reply:
Oh I just noticed that apparently this device binding isn't disposable which is wrong as the SPIDevice it holds should be disposed from here. Do you want to address that on this PR as well as it should be a straight forward change? If not I can add a follow up to do that.

Reply:
> Oh I just noticed that apparently this device binding isn't disposable which is wrong as the SPIDevice it holds should be disposed from here. Do you want to address that on this PR as well as it should be a straight forward change? If not I can add a follow up to do that.

Let's add a follow up for it. Thanks!